server/tests/src/test/java/org/infinispan/server/persistence/JdbcStringBasedCacheStorePassivationEnabled.java

```diff
@@ -0,0 +1,158 @@
+package org.infinispan.server.persistence;
+
+import static org.infinispan.server.security.Common.sync;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestResponse;
+import org.infinispan.client.rest.configuration.RestClientConfigurationBuilder;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.server.test.core.category.Persistence;
+import org.infinispan.server.test.core.persistence.Database;
+import org.infinispan.server.test.junit4.InfinispanServerRule;
+import org.infinispan.server.test.junit4.InfinispanServerTestMethodRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Tests String-based jdbc cache store under the following circumstances:
+ * <p>
+ * passivation == true --cache entries should get to the cache store only when evicted
+ * preload == false --after server restart, entries should NOT be preloaded to the cache
+ * purge == false --all entries should remain in the cache store after server restart
+ * (must be false so that we can test preload)
+ * <p>
+ * Other attributes like singleton, shared, fetch-state do not make sense in single node cluster.
+ *
+ */
+@Category(Persistence.class)
+@RunWith(Parameterized.class)
+public class JdbcStringBasedCacheStorePassivationEnabled {
+
+    @ClassRule
+    public static InfinispanServerRule SERVERS = PersistenceIT.SERVERS;
+
+    @Rule
+    public InfinispanServerTestMethodRule SERVER_TEST = new InfinispanServerTestMethodRule(SERVERS);
+
+    private final Database database;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        String[] databaseTypes = PersistenceIT.DATABASE.getDatabaseTypes();
+        List<Object[]> params = new ArrayList<>(databaseTypes.length);
+        for (String databaseType : databaseTypes) {
+            params.add(new Object[]{databaseType});
+        }
+        return params;
+    }
+
+    public JdbcStringBasedCacheStorePassivationEnabled(String databaseType) {
```

Comment:
I think we should consolidate the two `JdbcStringBasedCacheStorePassivation*` tests into a single `JdbcStringBasedCacheStore` test. We can then just use the method names to distinguish the preload enabled/disabled tests.